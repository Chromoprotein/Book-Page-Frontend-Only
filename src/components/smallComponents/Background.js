export default function Background({ children }) {
    return (
        <div className="bg-gradient-to-r from-sky-800 to-violet-800 min-w-full min-h-screen p-1">
            {children}
        </div>
    );
}