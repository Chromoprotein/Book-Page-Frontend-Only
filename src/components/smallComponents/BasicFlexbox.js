export default function BasicFlexbox({children}) {
    return (
        <div className="flex flex-col lg:flex-row flex-wrap flex-none justify-center place-content-center gap-3 p-2 w-full">
            {children}
        </div>
    );
}