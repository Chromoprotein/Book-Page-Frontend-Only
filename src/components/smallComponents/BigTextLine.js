export default function BigTextLine({children}) {
  return (
    <h2 className="text-white text-xl font-medium">{children}</h2>
  );
}