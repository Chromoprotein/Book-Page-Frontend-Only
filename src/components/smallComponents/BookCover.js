export default function BookCover({img}) {
    return (
        <img src={img} alt="Book cover" className="md:rounded-l-lg rounded-tr-lg rounded-tl-lg md:rounded-tr-none w-full md:h-full opacity-80 object-cover" />
    );
}