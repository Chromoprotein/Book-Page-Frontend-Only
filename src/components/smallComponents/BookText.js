import { TextLine } from "./TextLines";
import { BigTextLine } from "./TextLines";

export default function BookText({ series, title, author, year }) {

    const bookSeries = series ? <TextLine>{series}</TextLine> : "";

    return (
      <div className="px-6 pt-6 text-center">
        <BigTextLine>{title}</BigTextLine>
        {bookSeries}
        <TextLine> by {author}</TextLine>
        <TextLine> Read in {year}</TextLine>
      </div>
    );
}

