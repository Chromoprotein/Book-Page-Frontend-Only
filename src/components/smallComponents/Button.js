import IconContainer from "./IconContainer";

export default function Button({ children, buttonType, buttonOnClick, isDisabled, isActive, testId }) {

    return (
        <button type={buttonType} onClick={buttonOnClick} disabled={isDisabled} className={`rounded-full w-36 block mx-auto my-1 py-3 text-purple-600 font-semibold hover:text-white hover:bg-purple-800 hover:opacity-80 focus:bg-purple-800 focus:text-white disabled:bg-slate-700 disabled:text-slate-900  disabled:hover:opacity-100 ${isActive ? "bg-purple-800" : "bg-slate-900"}`} data-testid={testId}>
            <IconContainer>
                {children}
            </IconContainer>
        </button>
    );
}