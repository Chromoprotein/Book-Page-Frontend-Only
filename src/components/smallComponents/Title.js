import { Link } from "react-router-dom";
import logo from "../../logo.png";

export function Title() {
  return (
    <Link to={`/`} className="flex justify-center items-center pr-2">
      <img src={logo} className="w-20 pl-2" alt="Booksmosis logo"/>
      <h1 className="text-4xl opacity-100 font-medium tracking-wide font-sans text-slate-800">
        Booksmosis
      </h1>
    </Link>
  );
}