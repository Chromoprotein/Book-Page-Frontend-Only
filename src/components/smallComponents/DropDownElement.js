import { useState, useRef, useEffect } from "react";
import Button from "./Button";

export default function DropDownElement({ text, options, selectedOption, eventHandler }) {

    const [isOpen, setIsOpen] = useState(false);
    const dropdownRef = useRef(null);

    const toggleDropdown = () => setIsOpen(!isOpen);

    // Click outside the menu closes the menu
    // If the click target is not contained within dropdownRef.current, it's outside the menu
    useEffect(() => {
        const handleClickOutside = (event) => {
            if (dropdownRef.current && !dropdownRef.current.contains(event.target)) {
                setIsOpen(false);
            }
        }

        // Bind the event listener, listens to clicks outside the menu
        document.addEventListener("mousedown", handleClickOutside);
        return () => {
            // Unbind the event listener on clean up
            document.removeEventListener("mousedown", handleClickOutside);
        };
    }, [dropdownRef]);

  return (

    <div className="relative" ref={dropdownRef}>

    <Button buttonType="button" buttonOnClick={toggleDropdown}>{text}</Button>

    {isOpen && (
      <div className="fixed inset-0 md:inset-auto md:right-2 z-50 md:absolute md:z-10 md:mt-1 bg-white md:rounded-md shadow-lg w-screen md:w-32 h-fit">
        {options.map((option, index) => 
          <div 
            key={index} 
            className="hover:bg-purple-800 hover:text-white md:first:hover:rounded-t md:first:rounded-t md:last:hover:rounded-b md:last:rounded-b odd:bg-purple-200 px-4 py-2 h-20 md:h-12 flex justify-center items-center" 
            onClick={() => {
              eventHandler(option);
              setIsOpen(false);
            }}>
              {selectedOption === option ? <span>{option} &#10004;</span> : option}
            </div>
        )}
      </div>
    )}

    </div>

  );
}