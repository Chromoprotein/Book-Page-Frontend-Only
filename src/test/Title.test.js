import React from 'react';
import { render, screen } from '@testing-library/react';
import { Title } from '../components/smallComponents/Title';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

test('renders the "Booksmosis" title', () => {
      render(
        <BrowserRouter>
            <Title />
        </BrowserRouter>
    );
      const titleText = screen.getByText('Booksmosis');
      expect(titleText).toBeInTheDocument();
});

it('performs snapshot testing', () => {
  const tree = renderer.create(
    <BrowserRouter>
        <Title />
    </BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
