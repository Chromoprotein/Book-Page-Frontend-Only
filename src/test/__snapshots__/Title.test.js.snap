// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performs snapshot testing 1`] = `
<a
  className="flex justify-around items-center"
  href="/"
  onClick={[Function]}
>
  <img
    alt="Booksmosis logo"
    className="w-10 md:w-20 md:p-3"
    src="logo.png"
  />
  <h1
    className="hidden md:inline text-4xl opacity-100 font-medium tracking-wide font-sans text-slate-800"
  >
    Booksmosis
  </h1>
</a>
`;
